app/commands/exercise/representation/search.rb

```diff
@@ -0,0 +1,74 @@
+class Exercise::Representation::Search
+  include Mandate
+
+  # Use class method rather than constant for easier stubbing during testing
+  def self.requests_per_page = 20
+
+  initialize_with with_feedback: Mandate::NO_DEFAULT, criteria: nil, mentor: nil, track: nil,
+    order: :most_submissions, page: 1, paginated: true, sorted: true, only_mentored_solutions: false do
+    @order = order.try(&:to_sym)
+  end
+
+  def call
+    @representations = Exercise::Representation.includes(:exercise, :track)
+    filter_with_feedback!
+    filter_mentor!
+    filter_track!
+    filter_criteria!
+    filter_only_mentored_solutions!
+    sort! if sorted
+    paginate! if paginated
+    @representations
+  end
+
+  private
+  def filter_with_feedback!
+    if with_feedback
+      @representations = @representations.with_feedback
+    else
+      @representations = @representations.without_feedback
+    end
+  end
+
+  def filter_mentor!
+    return if mentor.blank?
+    return unless with_feedback
+
+    @representations = @representations.edited_by(mentor)
+  end
+
+  def filter_track!
+    return if track.blank?
+
+    @representations = @representations.for_track(track)
+  end
+
+  def filter_criteria!
+    return if criteria.blank?
+
+    exercise_ids = Exercise.where('title LIKE ?', "%#{criteria}%").
+      or(Exercise.where('slug LIKE ?', "%#{criteria}%")).
+      pluck(:id)
```

Comment:
I think we decided `select` was saner here, in case there are like 400 exercises that match `%a%` in which case mysql will blow up having that many things in an IN due to the sql being too long.